### PR TITLE
Use collect an HTTP quoted string

### PR DESCRIPTION
**Pull request by annevk**
See https://github.com/whatwg/fetch/pull/831.

***
<a href="https://whatpr.org/mimesniff/92.html" title="Last updated on Nov 22, 2018, 8:53 AM GMT (de107ef)">Preview</a> | <a href="https://whatpr.org/mimesniff/92/126286a...de107ef.html" title="Last updated on Nov 22, 2018, 8:53 AM GMT (de107ef)">Diff</a>

### CODE REVIEW

File: mimesniff.bs [hunk 1]
```diff
@@ -271,55 +269,26 @@ a <a>valid MIME type string</a> that does not contain U+003B (;).
...
     then:
 
     <ol>
-     <li><p>Advance <var>position</var> to the next <a>code point</a> in <var>input</var>.
+     <li><p>Let <var>unneeded</var> be null.
 
      <li>
-      <p>While true:
+      <p>Set <var>parameterValue</var> and <var>unneeded</var> to the result of
+      <a>collecting an HTTP quoted string</a> from <var>input</var>, given <var>position</var>.
```
Comment (domenic) [inline mimesniff.bs]: Maybe Fetch should have a wrapper algorithm that does the "unneeded" stuff for you?

**Comment by annevk** [comment]
I ended up adding a flag to the collecting algorithm so you it can have a single return value. It looks reasonably clean now.

File: mimesniff.bs [hunk 2]
```diff
@@ -271,55 +269,21 @@ a <a>valid MIME type string</a> that does not contain U+003B (;).
...
-          </ol>
-
-         <li><p>Otherwise, append U+005C (\) to <var>parameterValue</var> and
-         <a for=iteration>break</a>.
-        </ol>
-
-       <li><p>Otherwise, <a for=iteration>break</a>.
-      </ol>
+     <li><p>Set <var>parameterValue</var> to the result of <a>collecting an HTTP quoted string</a>
+     from <var>input</var>, given <var>position</var> and the <var>extract-value flag</var>.
```
Comment (domenic) [inline mimesniff.bs]: "var" is a bit weird here (e.g. that variable was never declared). But we don't have a good story for flags anyway, so I don't know what to really recommend.

Comment (annevk) [reply mimesniff.bs]: Previously you recommended `<var>` when I was using `<i>` for them.

Comment (domenic) [reply mimesniff.bs]: At the definition site that makes sense, but at the call site it's weird. But the whole setup has no obviously good answers so anything goes for now.

**Comment by annevk** [comment]
Note to self: this can likely land tomorrow once Fetch is reindexed.

**Pull request merged**